src/components/profile/connections/ConnectionBox.vue

```diff
@@ -0,0 +1,84 @@
+<template>
+  <div class="connection-box" :style="{ 'background-color': color }">
+<div class="header">
+  <img draggable="false" :class="{ 'account-profile': true, round: data.connected }" :src="data.connected ? data.account.avatar : getIcon(icon)" />
+  <div class="account-info">
+    <span class="account-name">{{ data.connected ? data.account.user : displayName }}</span>
+    <br>
+    <span v-if="data.connected" class="connection-name">{{ displayName }}</span>
+  </div>
+  <!-- TODO: Fix button to the right -->
+  <button v-if="!data.connected" class="button is-pulled-right" @click="openConnectWindow(data)">CONNECT</button>
+</div>
+    <div v-if="data.connected" class="content">
+      <slot class="config"></slot>
+    </div>
+  </div>
+</template>
+
+<script>
+export default {
+  name: 'ConnectionBox',
+  props: ['color', 'displayName', 'connected', 'icon', 'data'],
+  methods: {
+    openConnectWindow (data) {
+      this.$parent.$parent.openConnectWindow(data)
+    },
+    getIcon (icon) {
```

Comment:
This is wrong. The files I'll not be builded with the app.
Import all the icons before and then use a map object to return them.

And at the `icon` prop validation, you can check if the icon exists or not.

Reply:
You can use the svg-vue-loader to use svg as components, and then change your code to be

```vue
<Component :is="icon" />
```

you _only_ have to be sure that you register the component as the same icon key, or use the map object to map the key to the right component.

Reply:
@LucasLeandro1204 how can i do it? im using `svg-vue-loader` and on `LastfmConnection` im using this:
```
...
    <ConnectionBox
      :icon="LastfmIcon"
...
<script>
/* eslint-disable vue/no-unused-components */
import LastfmIcon from '../../../assets/logos/lastfm.svg'
...
```
but this doesnt work and gives this error:
 ```
Property or method "LastfmIcon" is not defined on the instance but referenced during render. Make sure that this property is reactive, either in the data option, or for class-based components, by initializing the property.
```

Reply:
ok im using static